Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -337,11 +337,120 @@ public struct TLSConfiguration {
+    /// contexts, you should use `forServer` instead.
+    ///
+    /// For customising fields, modify the returned TLSConfiguration object.
+    public static func forClient() -> TLSConfiguration {
+        return TLSConfiguration(cipherSuites: defaultCipherSuites,
+                                verifySignatureAlgorithms: nil,
+                                signingSignatureAlgorithms: nil,
+                                minimumTLSVersion: .tlsv1,
+                                maximumTLSVersion: nil,
+                                certificateVerification: .fullVerification,
+                                trustRoots: .default,
+                                certificateChain: [],
+                                privateKey: nil,
+                                applicationProtocols: [],
+                                shutdownTimeout: .seconds(5),
+                                keyLogCallback: nil,
+                                renegotiationSupport: .none,
+                                additionalTrustRoots: [])
+    }
+
+    /// Create a TLS configuration for use with server-side contexts.
+    ///
+    /// This provides sensible defaults while requiring that you provide any data that is necessary
+    /// for server-side function. For client use, try `forClient` instead.
+    ///
+    /// For customising fields, modify the returned TLSConfiguration object.
+    public static func forServer(
+        certificateChain: [NIOSSLCertificateSource],
+        privateKey: NIOSSLPrivateKeySource
+    ) -> TLSConfiguration {
+        return TLSConfiguration(cipherSuites: defaultCipherSuites,
+                                verifySignatureAlgorithms: nil,
+                                signingSignatureAlgorithms: nil,
+                                minimumTLSVersion: .tlsv1,
+                                maximumTLSVersion: nil,
+                                certificateVerification: .none,
+                                trustRoots: .default,
+                                certificateChain: certificateChain,
+                                privateKey: privateKey,
+                                applicationProtocols: [],
+                                shutdownTimeout: .seconds(5),
+                                keyLogCallback: nil,
+                                renegotiationSupport: .none,
+                                additionalTrustRoots: [])
+    }
+}
+
+// MARK: Deprecated constructors.
+
+extension TLSConfiguration {
     /// Create a TLS configuration for use with server-side contexts. This allows setting the `NIOTLSCipher` property specifically.
     ///
     /// This provides sensible defaults while requiring that you provide any data that is necessary
     /// for server-side function. For client use, try `forClient` instead.
+    @available(*, deprecated, renamed: "TLSConfiguration.forServer(certificateChain:privateKey:)")
```

Comment:
Is this right (and elsewhere)? I _think_ we want just `"forServer(certificateChain:privateKey:)"` here otherwise Xcode shoves in the `TLSConfiguration` as well when you apply the fix its.